solr/core/src/test/org/apache/solr/handler/admin/CoreAdminHandlerActionTest.java

```diff
@@ -0,0 +1,164 @@
+            "action1",
+            new CoreAdminHandlerTestAction1(),
+            "action2",
+            new CoreAdminHandlerTestAction2()));
+  }
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void testAction1() throws Exception {
+
+    SolrQueryResponse response = new SolrQueryResponse();
+
+    admin.handleRequestBody(req(CoreAdminParams.ACTION, "action1"), response);
+
+    Map<String, Object> actionResponse = ((NamedList<Object>) response.getResponse()).asMap();
+
+    assertTrue(
+        String.format(
+            Locale.ROOT,
+            "Action response should contain %s property",
+            CoreAdminHandlerTestAction1.PROPERTY_NAME),
+        actionResponse.containsKey(CoreAdminHandlerTestAction1.PROPERTY_NAME));
+    assertEquals(
+        String.format(
+            Locale.ROOT,
+            "Action response should contain %s value for %s property",
+            CoreAdminHandlerTestAction1.PROPERTY_VALUE,
+            CoreAdminHandlerTestAction1.PROPERTY_NAME),
+        CoreAdminHandlerTestAction1.PROPERTY_VALUE,
+        actionResponse.get(CoreAdminHandlerTestAction1.PROPERTY_NAME));
+  }
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void testAction2() throws Exception {
+
+    SolrQueryResponse response = new SolrQueryResponse();
+
+    admin.handleRequestBody(req(CoreAdminParams.ACTION, "action2"), response);
+
+    Map<String, Object> actionResponse = ((NamedList<Object>) response.getResponse()).asMap();
+
+    assertTrue(
+        String.format(
+            Locale.ROOT,
+            "Action response should contain %s property",
+            CoreAdminHandlerTestAction2.PROPERTY_NAME),
+        actionResponse.containsKey(CoreAdminHandlerTestAction2.PROPERTY_NAME));
+    assertEquals(
+        String.format(
+            Locale.ROOT,
+            "Action response should contain %s value for %s property",
+            CoreAdminHandlerTestAction2.PROPERTY_VALUE,
+            CoreAdminHandlerTestAction2.PROPERTY_NAME),
+        CoreAdminHandlerTestAction2.PROPERTY_VALUE,
+        actionResponse.get(CoreAdminHandlerTestAction2.PROPERTY_NAME));
+  }
+
+  @Test
+  public void testUnregisteredAction() throws Exception {
+
+    SolrQueryResponse response = new SolrQueryResponse();
+
+    assertThrows(
+        "Attempt to execute unregistered action should throw SolrException",
+        SolrException.class,
+        () -> admin.handleRequestBody(req(CoreAdminParams.ACTION, "action3"), response));
+  }
+
+  public static String randomString() {
+
+    int leftBound = 97; // a
+    int rightBound = 122; // z
+    int length = random().nextInt(6) + 5;
+
+    String randomString =
+        random()
+            .ints(leftBound, rightBound + 1)
+            .limit(length)
+            .collect(
+                () -> new StringBuilder(),
+                (item, value) -> item.appendCodePoint(value),
+                (item1, item2) -> item1.append(item2))
+            .toString();
+
+    return randomString;
+  }
```

Comment:
Maybe similar method is already implemented in some test util class I can use?

Reply:
I decided to remove it completely and switch to use simple `"property" + random().nextInt()`.